designer/server/src/main/scala/pl/touk/nussknacker/ui/additionalconfig/AdditionalUIConfigProvider.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.ui.additionalconfig
+
+import pl.touk.nussknacker.engine.api.component.{
+  ComponentGroupName,
+  ComponentId,
+  ParameterConfig,
+  ScenarioPropertyConfig,
+  SingleComponentConfig
+}
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+import pl.touk.nussknacker.engine.definition.component.parameter.editor.EditorExtractor
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{
+  ParameterValueCompileTimeValidation,
+  ValueInputWithFixedValues
+}
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.ui.additionalconfig.AdditionalUIConfigProvider.AdditionalUIConfig
+
+/**
+ * Trait allowing the provision of UI configuration for components and scenario properties, without requiring a model reload.
+ *
+ * TODO: The current implementation allows providing configs only for standard components - meaning that base components aren't handled.
+ */
+trait AdditionalUIConfigProvider extends Serializable {
+
+  def getAllForProcessingType(processingType: String): Map[ComponentId, AdditionalUIConfig]
+
+  def getScenarioPropertiesUIConfigs(processingType: String): Map[String, ScenarioPropertyConfig]
+
+}
+
+object AdditionalUIConfigProvider {
+  val empty = new DefaultAdditionalUIConfigProvider(Map.empty, Map.empty)
+
+  case class AdditionalUIConfig(
+      parameterConfigs: Map[String, ParameterAdditionalUIConfig],
+      icon: Option[String] = None,
+      docsUrl: Option[String] = None,
+      componentGroup: Option[ComponentGroupName] = None,
+      disabled: Boolean = false
+  ) {
+
+    def toSingleComponentConfigWithoutValidators: SingleComponentConfig =
+      SingleComponentConfig(
+        params = Some(parameterConfigs.mapValuesNow(_.toParameterConfigWithoutValidators)),
+        icon = icon,
+        docsUrl = docsUrl,
+        componentGroup = componentGroup,
+        disabled = disabled,
+        componentId = None
+      )
+
+  }
+
+  case class ParameterAdditionalUIConfig(
+      required: Boolean = false,
+      initialValue: Option[FixedExpressionValue],
+      hintText: Option[String],
+      valueEditor: Option[ValueInputWithFixedValues],
+      valueCompileTimeValidation: Option[ParameterValueCompileTimeValidation],
+  ) {
+
+    def toParameterConfigWithoutValidators: ParameterConfig = ParameterConfig(
```

Comment:
I see two options:
1. We keep `ParameterAdditionalUIConfig.valueCompileTimeValidation`, remove `WithoutValidators` from method names and left only a comment that currently it doesn't work - pros: there is a change, that this API won't change when we add the support for validators, cons: we expose some API that looks like support sth, but it doesn't
2. We remove `ParameterAdditionalUIConfig.valueCompileTimeValidation` and remove `WithoutValidators` from method names. Pros/cons: the same as above but reversed.

The current approach is sth in the middle, but only with cons

Reply:
ok, I think I prefer the first method